app/helpers/users_helper.rb

```diff
@@ -50,43 +50,24 @@ def user_soft_delete_confirm_message(user)
       (I18n.t("users.soft_delete_confirm_message.relatives", count: user.relatives.active.count) if user.relatives.active.any?),
     ].select(&:present?).join("\n\n")
   end
-end
-
-class DisplayableUser
-  include UsersHelper
-
-  delegate :first_name, :last_name, :birth_name, :address, :affiliation_number, :number_of_children, to: :user
-
-  attr_reader :user
-
-  def initialize(user, organisation)
-    @user = user
-    @user_profile = @user.profile_for(organisation)
-  end
-
-  def birth_date
-    birth_date_and_age(@user)
-  end
-
-  def caisse_affiliation
-    User.human_enum_name(:caisse_affiliation, @user.caisse_affiliation)
-  end
 
-  def family_situation
-    User.human_enum_name(:family_situation, @user.family_situation)
-  end
+  def users_to_sentence(users, current_agent, current_user)
```

Comment:
je pense que c'est une mauvaise idée de passer explicitement current_agent et current_user en parametres ici : 
- ces methodes `current_x` sont elles memes des helpers fournies par devise
- les helpers peuvent appeler d'autres helpers. la valeur sera toujours la meme qu'ils soient appelés depuis la vue ou depuis le helper
- tel quel, avec les memes noms, c'est perturbant parce que le nom de la variable cache le nom du helper
- si la raison pour laquelle tu as fait ca c'est que c'est plus simple a tester, je pense qu'il doit y avoir une solution moins genante comme mocker le resultat des methodes `current_x`

Reply:
C'est en parti pour les tests, mais surtout parce que je préfère m'approcher de vrai fonction, en réduisant les dépendances...
Tu as raison sur le fait qu'utiliser les même nom, c'est idiot !

Pour le reste, effectivement, j'ai eu une approche par les tests, et ça m'a amené à faire autrement. Ne pas utiliser `current_agent()` et `current_user()`, je crois qu'il y a plus ou moins à voir avec le I et le D des principes SOLID (tu connais ?)

Ceci étant, si c'est plus simple pour toi, je peux remettre le current_agent current_user (pas sur d'avoir envie de mettre en mock par contre :thinking: )

Reply:
je vois l'idée d'injection des dependances oui je comprends le principe mais je ne suis pas convaincu que ce soit adapté ici. je voterais plus pour laisser cette dépendance :
- les current_xxx sont appelés a beaucoup d'autres endroits dans les helpers, ca serait un peu le seul qui fait différrement 
- globablement les helpers sont pas mal interdépendants, par exemple ils utilisent les path helpers. est-ce que ca veut dire qu'on devrait la aussi injecter manuellement les path helpers pour chacun ? ca me parait pas tres "The Rails Way™️"

mais je ne m'y opposerais pas du tout c'est du nitpick ici, si tu preferes en injectant les dependances ici c'est ok ! 

le mock c'etait completement une proposition tu en fais ce que tu veux ! c'est comme ca que j'aurais testé la methode je pense, je ne connais pas trop d'autres maniere.

Reply:
Je crois que ce réfactoring était déjà très en bordure de la PR... je vais revenir à un truc plus classique, et on verra plus tard pour reprendre ce sujet.

Reply:
(et pour info, je pense que la rails way n'est pas toujours très aligné avec ce que je pense être du « beau code » ou du « code propre© » :D On pourra en reparler)